pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -590,6 +590,106 @@ func createSplitFeltHinter(resolver hintReferenceResolver) (hinter.Hinter, error
 	return newSplitFeltHint(low, high, value), nil
 }
 
+func newSignedDivRemHint(value, div, bound, r, biased_q hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SignedDivRem",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.math_utils import as_int, assert_integer
+			// assert_integer(ids.div)
+			// assert 0 < ids.div <= PRIME // range_check_builtin.bound, f'div={hex(ids.div)} is out of the valid range.'
+			// assert_integer(ids.bound)
+			// assert ids.bound <= range_check_builtin.bound // 2, f'bound={hex(ids.bound)} is out of the valid range.'
+			// int_value = as_int(ids.value, PRIME)
+			// q, ids.r = divmod(int_value, ids.div)
+			// assert -ids.bound <= q < ids.bound, f'{int_value} / {ids.div} = {q} is out of the range [{-ids.bound}, {ids.bound}).'
+			// ids.biased_q = q + ids.bound
+
+			//> assert_integer(ids.div)
+			divFelt, err := hinter.ResolveAsFelt(vm, div)
+			if err != nil {
+				return err
+			}
+			//> assert 0 < ids.div <= PRIME // range_check_builtin.bound, f'div={hex(ids.div)} is out of the valid range.'
+			if divFelt.IsZero() || !utils.FeltLe(divFelt, &utils.PrimeHigh) {
+				return fmt.Errorf("div=%v is out of the valid range.", divFelt)
+			}
+
+			//> assert_integer(ids.bound)
+			boundFelt, err := hinter.ResolveAsFelt(vm, bound)
+			if err != nil {
+				return err
+			}
+			//> assert ids.bound <= range_check_builtin.bound // 2, f'bound={hex(ids.bound)} is out of the valid range.'
+			if !utils.FeltLe(boundFelt, &utils.Felt127) {
+				return fmt.Errorf("bound=%v is out of the valid range.", boundFelt)
+			}
+			//> int_value = as_int(ids.value, PRIME)
+			valueFelt, err := hinter.ResolveAsFelt(vm, value)
+			if err != nil {
+				return err
+			}
+
+			var divBig, boundBig big.Int
+			divFelt.BigInt(&divBig)
+			boundFelt.BigInt(&boundBig)
+
+			// int_value = as_int(ids.value, PRIME)
+			intValueBig := AsInt(valueFelt)
+			//> q, ids.r = divmod(int_value, ids.div)
+			qBig, rBig := new(big.Int).DivMod(&intValueBig, &divBig, new(big.Int))
+			rFelt := new(fp.Element).SetBigInt(rBig)
+			//> assert -ids.bound <= q < ids.bound, f'{int_value} / {ids.div} = {q} is out of the range [{-ids.bound}, {ids.bound}).'
+			if !(qBig.Cmp(new(big.Int).Neg(&boundBig)) >= 0 && qBig.Cmp(&boundBig) == -1) {
+				return fmt.Errorf("%v / %v = %v is out of the range [-%v, %v].", valueFelt, divFelt, qBig, boundFelt, boundFelt)
+			}
+
+			rAddr, err := r.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			rValue := memory.MemoryValueFromFieldElement(rFelt)
+			err = vm.Memory.WriteToAddress(&rAddr, &rValue)
+			if err != nil {
+				return err
+			}
```

Comment:
Same as before, I'd move these lines up, right after the declaration of `rFelt`, as writing the value to memory is still something you have to do from that step

Reply:
done